### DISCUSSION TITLE
How can I create a docker image for a platform other than amd64?

### DISCUSSION THREAD

**Discussion by arulagrawal** [concourse/concourse#7285] [Help & Support]
I have concourse up and running, and by following https://concourse-ci.org/building-and-pushing-an-image.html I have been able to automatically containerize and publish to docker hub my simple golang program (1 source file, so super small). I'm trying to figure out how I can make concourse also make an arm64 image and publish it under the same tag, so that my image can be multi-arch.

I know cross-compiling is easy with go, but how can I incorporate this into concourse?

I wasn't able to find how to set the docker manifest(?)

Thanks

**Comment by taylorsilva** [comment]
I don't think the `oci-build-task` supports this workflow currently. Sounds like a feature request for that.

There are many other ways to build images though, so you could pick another tool to create your images for you, one that meets this requirement.

**Reply by taylorsilva** [reply]
played around with buildkit a bit and looks like you need to pass in `--opt platform=` to specify the platform for the image. Might be kinda easy to add to the oci-build-task.

```
sudo buildctl build --frontend dockerfile.v0 --opt platform=linux/arm64 --local context=. --local dockerfile=. --output type=oci > image.tar
```

**Reply by taylorsilva** [reply]
It was quite easy! https://github.com/concourse/oci-build-task/pull/66

**Reply by arulagrawal** [reply]
Cool! Guess I'll wait till this is pushed out to a release!

**Reply by taylorsilva** [reply]
It's out https://github.com/concourse/oci-build-task/releases/tag/v0.9.0
